UPGRADE-3.x.md

```diff
@@ -1,6 +1,14 @@
 UPGRADE 3.x
 ===========
 
+`Sonata\AdminBundle\Admin\AdminExtensionInterface` has been deprecated and segregated into single
+method interfaces, they can be found in `Sonata\AdminBundle\Admin\Extension` namespace.
+
+`Sonata\AdminBundle\Admin\AbstractAdminExtension` has been deprecated without alternative.
+
+`Sonata\AdminBundle\Admin\AbstractAdmin::addExtension` has been deprecated, use
+`Sonata\AdminBundle\Admin\AbstractAdmin::addLightweightExtension` instead.
```

Comment:
Oh right, that name… we have to find something better.

Reply:
Hook? Listener?

Reply:
`addAdminExtension`?